tools/defaults.bzl

```diff
@@ -79,6 +81,47 @@ def pkg_npm(name, use_prodmode_output = False, **kwargs):
         deps = deps,
     )
 
+    # Merge package.json with root package.json and perform various substitutions to
+    # prepare it for release.
+    jq(
+        name = "basic_substitutions",
+        srcs = ["//:package.json"] + kwargs.pop("srcs"),
+        filter = """
```

Comment:
Out of curiosity, why don't we use a Bash/Node script for merging the package json? That would seem a little more readable and maintainable IMO, plus it wouldn't rely on additional dependencies like `jq` binaries.

Reply:
Writing a rule that invokes a custom script would've been a viable option, too. We decided to build the `jq` rule to solve the problem of json substitutions more generally, so that others can use it and perform different types of substitutions. Jq filters do require some learning, but the library is also fairly popular and widespread.

If you prefer a custom script I can do that instead. Alternatively I could throw this filter into a file to reduce verboseness—jq can load filters from files as well.

Reply:
Yeah, I don't feel strongly enough and I don't think I can make a decision here, but generally I'd prefer to stick with technology that is already commonly used in e.g. this repo. Jq js awesome, but a JS script may be more flexible (in the future), and readable (plus it removes an additional dependency)

We can just leave as is, and move forward (from my side). Happy holidays!

Reply:
Happy holidays!

@josephperrott how do you feel about using `jq` here versus a custom js script?